Marlin/Marlin_main.cpp

```diff
@@ -3576,10 +3576,28 @@ inline void gcode_M115() {
  * M117: Set LCD Status Message
  */
 inline void gcode_M117() {
-  char* codepos = strchr_pointer + 5;
+  millis_t codenum = 0;
+  char* codepos;
+  if (code_seen('S')) {
```

Comment:
I just copied G4, but dropped P because showing messages for milliseconds is meaningless.
If You get #1966 to work i'd prefer yours.